dbms/include/DB/Functions/FunctionsGeo.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include <DB/DataTypes/DataTypesNumberFixed.h>
+#include <DB/Functions/IFunction.h>
+#include <ext/range.hpp>
+#include <math.h>
+
+#define D_R (M_PI / 180.0)
+
+namespace DB{
+
+const Float64 EARTH_RADIUS_IN_METERS = 6372797.560856;
+
+static inline Float64 deg_rad(Float64 ang) { return ang * D_R; }
+static inline Float64 rad_deg(Float64 ang) { return ang / D_R; }
+  
+class FunctionGeoDistance : public IFunction
```

Comment:
You should add comments about what this function doing.

It's unclear, in what units it returns result. If it is unspecified relative units - it's Ok, but we need comment about it.

Also, it's not clear, what it is doing if passed arguments are out of range.